File: python/1.py
class Solution:
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        m = {}
        for i, n in enumerate(nums):
            m[n] = i
        
        for i, n in enumerate(nums):
            if target - n in m and m[target - n] != i:
                return [m[target - n], i]

File: golang/1.go
func twoSum(nums []int, target int) []int {
    m := make(map[int]int)
    for i, n := range nums {
        m[n] = i
    }
    for i, n := range nums {
        if j, ok := m[target - n]; ok {
            if j != i {
                return []int{i, j}
            }
        }
    }
    return []int{}
}